hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/scm/protocol/ScmBlockLocationProtocol.java

```diff
@@ -59,8 +60,30 @@
    * @return allocated block accessing info (key, pipeline).
    * @throws IOException
    */
-  List<AllocatedBlock> allocateBlock(long size, int numBlocks,
+  @Deprecated
+  default List<AllocatedBlock> allocateBlock(long size, int numBlocks,
       ReplicationType type, ReplicationFactor factor, String owner,
+      ExcludeList excludeList) throws IOException {
+    return allocateBlock(size, numBlocks, ReplicationConfig
+        .fromTypeAndFactor(type, factor), owner, excludeList);
+  }
+
+  /**
+   * Asks SCM where a block should be allocated. SCM responds with the
+   * set of datanodes that should be used creating this block.
+   *
+   * @param size              - size of the block.
+   * @param numBlocks         - number of blocks.
+   * @param replicationConfig - replicationConfiguration
+   * @param owner             - service owner of the new block
+   * @param excludeList       List of datanodes/containers to exclude during
+   *                          block
+   *                          allocation.
+   * @return allocated block accessing info (key, pipeline).
+   * @throws IOException
+   */
+  List<AllocatedBlock> allocateBlock(long size, int numBlocks,
```

Comment:
Why owner fix included in this patch?

Reply:
Only javadoc fixes are included. Here it is showed only because I modified ReplicationConfig at the same line:

Old code
```
       ReplicationType type, ReplicationFactor factor, String owner,
``` 

New code:

```
       ReplicationType type, ReplicationFactor factor, String owner,
```